block/src/Pos/DB/Block.hs

```diff
@@ -107,25 +105,21 @@ putSerializedBlunds
     :: (HasConfiguration, MonadRealDB ctx m, MonadDB m)
     => NonEmpty SerializedBlund -> m ()
 putSerializedBlunds (toList -> bs) = do
-    bdd <- view blockDataDir <$> getNodeDBs
-    let allData = map (\(b,u) -> let (dP, bP, uP) = getAllPaths bdd (headerHash b)
-                                 in (dP,(b,u,bP,uP))
+    let allData = map (\(b,u) -> let hh = formatToString hashHexF (headerHash b)
+                                 in (serialize' b, unSerialized u, hh, undoModifier hh)
                       )
                       bs
-    forM_ (ordNub $ map fst allData) $ \dPath ->
-        liftIO $ createDirectoryIfMissing False dPath
-    forM_ (map snd allData) $ \(blk,serUndo,bPath,uPath) -> do
-        putData bPath blk
-        putRawData uPath (unSerialized serUndo)
+    DB{..} <- getBlockDataDB
+    forM_ allData $ \(blk,serUndo,bkey,ukey) -> liftIO $ do
+        Rocks.put rocksDB rocksWriteOpts (fromString bkey) blk
+        Rocks.put rocksDB rocksWriteOpts (fromString ukey) serUndo
     putHeadersIndex $ toList $ map (CB.getBlockHeader . fst) bs
 
 deleteBlock :: (MonadRealDB ctx m, MonadDB m) => HeaderHash -> m ()
 deleteBlock hh = do
     deleteHeaderIndex hh
-    bdd <- view blockDataDir <$> getNodeDBs
-    let (_, bPath, uPath) = getAllPaths bdd hh
-    deleteData bPath
-    deleteData uPath
+    DB{..} <- getBlockDataDB
+    Rocks.delete rocksDB rocksWriteOpts (fromString . formatToString hashHexF $ hh)
```

Comment:
AFAIU, previous implementation deleted block and undo, but this one deletes only block.